Marlin/src/module/settings.cpp

```diff
@@ -2855,11 +2814,11 @@ void MarlinSettings::reset() {
       #define PID_DEFAULT(N,E) DEFAULT_##N
     #endif
     HOTEND_LOOP() {
-      PID_PARAM(Kp, e) =      float(PID_DEFAULT(Kp, ALIM(e, defKp)));
-      PID_PARAM(Ki, e) = scalePID_i(PID_DEFAULT(Ki, ALIM(e, defKi)));
-      PID_PARAM(Kd, e) = scalePID_d(PID_DEFAULT(Kd, ALIM(e, defKd)));
-      TERN_(PID_EXTRUSION_SCALING, PID_PARAM(Kc, e) = float(PID_DEFAULT(Kc, ALIM(e, defKc))));
-      TERN_(PID_FAN_SCALING, PID_PARAM(Kf, e) = float(PID_DEFAULT(Kf, ALIM(e, defKf))));
+      heater.pid.setKp(float(PID_DEFAULT(Kp, ALIM(e, defKp))));
+      heater.pid.setKi(scalePID_i(PID_DEFAULT(Ki, ALIM(e, defKi))));
+      heater.pid.setKd(scalePID_d(PID_DEFAULT(Kd, ALIM(e, defKd))));
+      TERN_(PID_EXTRUSION_SCALING, heater.pid.setKc(float(PID_DEFAULT(Kc, ALIM(e, defKc)))));
+      TERN_(PID_FAN_SCALING, heater.pid.setKf(float(PID_DEFAULT(Kf, ALIM(e, defKf)))));
```

Comment:
Note that without `DISTINCT_E_FACTORS` all E heaters should share the same data, and no loop should be required.

Reply:
This is tough to implement. The basic idea of this PR part is to have all heater-like object be instance of a common Heater class. So even if they all share the same value, they are independent instances. What you are asking for is to use a kind of reference/pointer to the *single* value vectors here. It's possible, but it's not done now. Probably in a later PR.